brownie/network/rpc.py

```diff
@@ -352,6 +353,8 @@ def revert(self) -> str:
         self._undo_buffer.clear()
         self._redo_buffer.clear()
         self._snapshot_id = self._current_id = self._revert(self._snapshot_id)
+        for project in brownie.project.get_loaded_projects():
```

Comment:
This feels out of place to me.  In general, I think `project` should be importing and interacting with `network`, but not the other way around.

Maybe instead add `_revert` and `_reset` methods to `Project`, and call `rpc._register_revert` during `Project.load`?  And then handle the `_clear_dev_deployments` logic in those functions.